### ISSUE TITLE
Bug Report: Phone numbers

### ISSUE THREAD

**Issue by Ribeccalin** [mifosio/fims-web-app#14]
When creating a customer, the phone number is not required to be a number:
![screenshot 101](https://cloud.githubusercontent.com/assets/28103295/25524932/1794eba4-2bd2-11e7-995f-0391ba929893.png)

**Comment by DanCarl857** [comment]
Hello @crain I would like to work on this issue. I forked the repo for this web app and have run it. I am using tenant: default, username: mifos and password: password but I get an error message. Could you please help me with how to login to the app?

**Comment by crain** [comment]
@DanCarl857 - you need to clone the demo server repository here: https://github.com/mifosio/demo-server. Follow the instructions and if its running you should be able to login with tenant: playground and the credentials here: 

https://github.com/mifosio/demo-server/blob/develop/src/main/java/io/mifos/dev/ServiceRunner.java#L326

Let me know if that worked for you.

**Comment by DanCarl857** [comment]
Thanks very much @crain. Let me do that

**Comment by DanCarl857** [comment]
Hi @crain I ran the demo server and it fails with the following stack trace: https://paste.kde.org/ppoueku7b
Apparently it is not finding some of the other services. Am I supposed to run those too? And if yes, how am I supposed to do that. Sorry for the inconveniences

**Comment by crain** [comment]
Yes you need to build and run all services locally as we currently have no public artifacts available.

As an example you can use this batch script here:
https://github.com/mifosio/demo-server/blob/develop/scripts/windows/initial-setup.bat

This script takes your cloned repositories, pulls them, executes a build and publishes the artifact into your local repository afterwards.

**Comment by DanCarl857** [comment]
Thanks @crain. For the script you sent could I would write the ones for Linux and Mac users and send as a PR because I am on a Mac

**Comment by DanCarl857** [comment]
Hi @crain ran and built all the services but the provisioner service fails with this [error](https://paste.kde.org/po27szb9k) and due to this, the demo server now fails with this [error](https://paste.kde.org/pppbcdv69). Some dependencies in the provisioner service can't be found. I used both the script I wrote and manually but I still get the same error.

**Comment by myrle-krantz** [comment]
Hey @DanCarl857 you also need to git and build permitted feign client. My apologies for not updating the build script you're using. I don't use it, so I missed it.

**Comment by DanCarl857** [comment]
Hi @myrle-krantz please where can I find the permitted feign client? I was writing the build scripts for *nix systems so I'll just add it to all the build scripts

**Comment by myrle-krantz** [comment]
https://github.com/mifosio/permitted-feign-client

**Comment by DanCarl857** [comment]
Thanks @myrle-krantz it works now. I updated the windows scripts and added the scripts for *nix systems. Now when I run the demo-server after all the builds, it begins properly(a few services run) and then I get this [log](https://paste.kde.org/pdayfklyo). Any idea how to fix this. From what I gather, the feign client is failing to make a certain POST request for some reason.

**Comment by myrle-krantz** [comment]
It looks like Cassandra isn't starting properly, and the application is failing to spin up.  Maybe you already have Cassandra started somewhere?

**Comment by crain** [comment]
@DanCarl857 - how do you start the demo-server?

If the problem still exists please create a new issue here: https://github.com/mifosio/demo-server/issues/new and we go from there.

**Comment by DanCarl857** [comment]
Hi @crain I managed to run the server but I keep getting an error on the cassandra service. @myrle-krantz suggested it was casssandra on my machine still running. I have been trying to stop that to no avail. Each time I stop it, it just relaunches itself. Still looking at other options on how to stop cassandra or worst case remove it

**Comment by DanCarl857** [comment]
But I already finished the scripts for *nix systems and sent a PR

**Comment by Ribeccalin** [comment]
Transferred to JIRA

**Comment by Ippezrobert** [comment]
@myrle-krantz , It will be best if all the prerequisites are listed with their download links for all platforms e.g Windows, Linux etc.
Am deploying in Windows Environment and provision service is failing with the error below:-

19:38:09.666 [main] WARN  o.eclipse.jetty.webapp.WebAppContext - Failed startup of context o.s.b.c.e.j.JettyEmbeddedWebAppContext@1734f68{/provisioner/v1,file:///C:/Users/Skyline%20Technologies/AppData/Local/Temp/jetty-docbase.6976753875901215218.2020/,UNAVAILABLE}
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'io.mifos.anubis.config.AnubisSecurityConfigurerAdapter': Unsatisfied dependency expressed through method 'configureGlobal' parameter 1; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'io.mifos.anubis.security.IsisAuthenticatedAuthenticationProvider': Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'io.mifos.anubis.provider.TenantRsaKeyProvider': Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'io.mifos.anubis.repository.TenantAuthorizationDataRepository': Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cassandraSessionProvider' defined in io.mifos.core.cassandra.config.CassandraConnectorConfiguration: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.mifos.core.cassandra.core.CassandraSessionProvider]: Factory method 'cassandraSessionProvider' threw exception; nested exception is com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /127.0.0.1:9142 (com.datastax.driver.core.exceptions.TransportException: [/127.0.0.1] Cannot connect))
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredMethodElement.inject(AutowiredAnnotationBeanPostProcessor.java:648)

It seems Cassandra is not detected running or something. But in my services, its running. Could it be i installed a wrong Cassandra.

This is what i installed DataStax Cassandra Community Server 3.0.9

**Comment by crain** [comment]
@Ippezrobert - I am assuming this is an issue with the demo-server you are trying to start. Please create an issue here: https://github.com/mifosio/demo-server/issues/new.

You can start the demo-server with embedded cassandra and mysql so you don't have to install cassandra and/or mariadb.

**Comment by Ippezrobert** [comment]
@crain , Thanks for your response. I will create an issue in https://github.com/mifosio/demo-server/issues/new.

Is there a specific command to start the demo-server with embedded cassandra and mysql? Am using this command to start the demo-server 
_java -jar -Ddemoserver.persistent=true demo-server-0.1.0-BUILD-SNAPSHOT.jar_

**Comment by crain** [comment]
@Ippezrobert, yes you can start with demoserver.persistent=false. All available parameter are described here: https://github.com/mifosio/demo-server.
